modules/policy/engine/src/main/java/org/eclipse/sw360/antenna/policy/engine/PolicyEngine.java

```diff
@@ -28,21 +28,21 @@ public PolicyEngine(final Collection<RuleExecutor> executors) {
         this.executors = executors;
     }
 
-    public Collection<ViolationIssue> evaluate (final Collection<ThirdPartyArtifact> thirdPartyArtifacts) {
+    public Collection<PolicyViolation> evaluate (final Collection<ThirdPartyArtifact> thirdPartyArtifacts) {
         LOGGER.info("Start policy engine run");
         LOGGER.debug("Artifacts are" + thirdPartyArtifacts.toString());
 
         return executors
                 .parallelStream()
                 .map(executor -> executor.executeRules(thirdPartyArtifacts))
-                .flatMap(results -> results.stream())
-                .collect(Collectors.toSet());
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
```

Comment:
Note: This is a subtle change to before, because violations are not de-duplicated here anymore.

Reply:
That is right, but I thought about it and I concluded that there are no duplications because every rule is executed on artifacts once, so two violations should always have a different context.